sdk/keyvault/keyvault-certificates/src/identifier.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { parseKeyvaultIdentifier } from "./core/utils";
+
+/**
+ * Valid collection names for Key Vault Certificate identifiers.
+ */
+export type KeyVaultCertificatesIdentifierCollectionName = "certificates" | "deletedcertificates";
+
+/**
+ * Represents a Key Vault identifier and its parsed contents.
+ */
+export interface ParsedKeyVaultCertificatesIdentifier {
+  /**
+   * The type of resource under Key Vault that this identifier is referring to.
+   */
+  collection: KeyVaultCertificatesIdentifierCollectionName;
+
+  /**
+   * The originally received identifier.
+   */
+  id: string;
+
+  /**
+   * The Key Vault Certificate unique identifier (an URl).
+   */
+  vaultUrl: string;
+
+  /**
+   * The version of Key Vault Certificate. Might be undefined.
+   */
+  version?: string;
+
+  /**
+   * The name of the Key Vault Certificate.
+   */
+  name: string;
+}
+
+/**
+ * Parser of the KeyVaultIdentifier for the Key Vault Certificates Client.
+ */
+export class KeyVaultCertificatesIdentifier implements ParsedKeyVaultCertificatesIdentifier {
+  /**
+   * The type of resource under Key Vault that this identifier is referring to.
+   */
+  collection: KeyVaultCertificatesIdentifierCollectionName;
+
+  /**
+   * The originally received identifier.
+   */
+  public id: string;
+
+  /**
+   * The Key Vault Certificate unique identifier (an URl).
+   */
+  public vaultUrl: string;
+
+  /**
+   * The version of Key Vault Certificate. Might be undefined.
+   */
+  public version?: string;
+
+  /**
+   * The name of the Key Vault Certificate.
+   */
+  public name: string;
+
+  /**
+   * Parses a Key Vault identifier.
+   * @param url Key Vault identifier
+   */
+  constructor(url: string) {
```

Comment:
since the only method in this class is the constructor, why not have a method that takes a URL and returns a `ParsedKeyVaultCertificatesIdentifier` instead of using a class to do this?

Reply:
This is to align with other languages. But let's get @heaths 's feedback

Reply:
Java and .NET can't have global methods (well, .NET soon will support it but nothing something we'll be able to use). I can see what Python and JavaScript might want to do this, though: it's idiomatic. For .NET, at least, this will be static `Parse` and `TryParse` methods on the class it returns, like `int.Parse`, etc. Java may use a builder, but will follow a similar naming strategy

Given that, it's probably a better question for your language architects. Do we already have global methods like you're proposing? Any concerns with discoverability? Etc.

/cc @chlowell @iscai-msft for Python

Reply:
I mean, a function is perfectly viable for TypeScript. I'll ask Brian just to have his seal of approval.

Reply:
@bterlson would a function or a class be better? knowing that only Python and TypeScript would be able to make this a function.

Reply:
Definitely a function.

Reply:
Thank you!